src/nbl/video/ISurfaceVK.cpp

```diff
@@ -0,0 +1,21 @@
+#include "nbl/video/surface/ISurfaceVK.h"
+
+#include "nbl/video/CVulkanPhysicalDevice.h"
+
+namespace nbl {
+namespace video
+{
+
+bool ISurfaceVK::isSupported(const IPhysicalDevice* dev, uint32_t _queueFamIx) const
+{
+    // TODO runtime check if vulkan physical device?
```

Comment:
yes please, everywhere you can even make a `CVulkanPhysicalDevice CVulkanPhysicalDevice::castDown(IPhysicalDevice* dev)` and similar everywhere that return do a debug break and return nullptr if the type doesnt match

Reply:
@achalpandeyy old comment about checking device compat